opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -900,6 +901,32 @@ namespace detail {
             // OPM_AD_DUMP(residual_.material_balance_eq[ Gas ]);
 
         }
+
+
+        if (param_.update_equations_scaling_) {
+            updateEquationsScaling();
+        }
+
+    }
+
+
+
+
+
+    template <class Grid, class Implementation>
+    void
+    BlackoilModelBase<Grid, Implementation>::updateEquationsScaling() {
+        ADB::V B;
+        const Opm::PhaseUsage& pu = fluid_.phaseUsage();
+        for ( int idx=0; idx<MaxNumPhases; ++idx )
+        {
+            if (active_[idx]) {
+                const int pos    = pu.phase_pos[idx];
+                const ADB& tempB = rq_[pos].b;
```

Comment:
These variable names are a bit confusing. I suggest you use temp_b (since it is little b, not capital B). Then B = 1/b as expected.